tools/firtool/firtool.cpp

```diff
@@ -959,6 +966,35 @@ static LogicalResult processBuffer(
   return success();
 }
 
+bool isFirLoc(Location loc) {
+  if (auto f = dyn_cast<FileLineColLoc>(loc))
+    return f.getFilename().strref().endswith(".fir");
```

Comment:
I don't love it.  But my only alternative was to make new location classes, which seems disruptive.

Reply:
Tried to do this recently for debug info. Turns out you can't have your own location classes. Would have to be upstream. We could use a `NamedLoc` though, and assign names like `"info"` or `"fir"` depending on where they came from.